scripts/o3de/o3de/android_support.py

```diff
@@ -1937,7 +1937,31 @@ def resolve_splash_overrides(self, az_android_dst_path, az_android_package_env):
         if not splash_overrides:
             return
 
-        orientation = az_android_package_env['ORIENTATION']
+        orientation_source = az_android_package_env['ANDROID_SCREEN_ORIENTATION']
+        orientation = ORIENTATION_LANDSCAPE
+
+        # Check orientation type and convert if needed
+        if isinstance(orientation_source, str):
+            if orientation_source not in ORIENTATION_MAPPING:
+                raise RuntimeError(
+                    f'Invalid orientation "{orientation}" in android_project.json. '
+                    f'Expected one of: {list(ORIENTATION_MAPPING.keys())}'
+                )
+            orientation = ORIENTATION_MAPPING[orientation_source]
+        elif isinstance(orientation_source, int):
+            VALID_ORIENTATIONS = {ORIENTATION_LANDSCAPE, ORIENTATION_PORTRAIT, ORIENTATION_ALL}
+            if orientation_source not in VALID_ORIENTATIONS:
+                raise RuntimeError(
+                    f'Invalid numeric orientation "{orientation}" in android_project.json. '
+                    f'Expected one of: {VALID_ORIENTATIONS}'
+                )
+            orientation = orientation_source
+        else:
+            raise RuntimeError(
```

Comment:
nit: Can we use [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError) here instead of `RuntimeError` ?